data/lua/stun.lua

```diff
@@ -0,0 +1,72 @@
+
+local _ = wesnoth.textdomain "wesnoth-units"
+local on_event = wesnoth.require("on_event")
+local old_unit_status = wesnoth.interface.game_display.unit_status
+
+function wesnoth.interface.game_display.unit_status()
+    local u = wesnoth.interface.get_displayed_unit()
+    if not u then return {} end
+    local s = old_unit_status()
+
+    if u.status.stunned then
+        table.insert(s, wml.tag.element{
+            image = "misc/stunned-status-icon.png",
+            tooltip = _"stunned: This unit is stunned. It cannot enforce its Zone of Control."
+        })
+    end
+
+    return s
+end
+
+local function on_hit(weapon, opponent)
+    if not weapon:matches{special_id = 'stun'} then return end
```

Comment:
Has this already been filtered for whether the ability is active or not? Although the Brawler's ability is always active, if `stun` becomes a general ability like `slow` then it should support abilities with filters.

Should it match by `id`, or should it use a `[stun]` tag and match by tagname?

Reply:
Yeah you're right, it should probably be `special_id_active`.

Reply:
Or actually I guess it should use a `[stun]` tag and `special_type_active`.